SwiftVoiceQuickstart/ViewController.swift

```diff
@@ -116,111 +113,160 @@ class ViewController: UIViewController, TVONotificationDelegate, TVOCallDelegate
     }
 
     @IBAction func mainButtonPressed(_ sender: Any) {
-        if let call = self.activeCall {
-            self.userInitiatedDisconnect = true
-            performEndCallAction(uuid: call.uuid)
-            self.toggleUIState(isEnabled: false, showCallControl: false)
-        } else {
-            let uuid = UUID()
-            let handle = "Voice Bot"
+        guard activeCall == nil else {
+            userInitiatedDisconnect = true
+            performEndCallAction(uuid: activeCall!.uuid)
+            toggleUIState(isEnabled: false, showCallControl: false)
             
-            self.checkRecordPermission { (permissionGranted) in
-                if (!permissionGranted) {
-                    let alertController: UIAlertController = UIAlertController(title: "Voice Quick Start",
-                                                                               message: "Microphone permission not granted",
-                                                                               preferredStyle: .alert)
-                    
-                    let continueWithMic: UIAlertAction = UIAlertAction(title: "Continue without microphone",
-                                                                       style: .default,
-                                                                       handler: { (action) in
-                        self.performStartCallAction(uuid: uuid, handle: handle)
-                    })
-                    alertController.addAction(continueWithMic)
-                    
-                    let goToSettings: UIAlertAction = UIAlertAction(title: "Settings",
-                                                                    style: .default,
-                                                                    handler: { (action) in
-                        UIApplication.shared.open(URL(string: UIApplicationOpenSettingsURLString)!,
-                                                  options: [UIApplicationOpenURLOptionUniversalLinksOnly: false],
-                                                  completionHandler: nil)
-                    })
-                    alertController.addAction(goToSettings)
-                    
-                    let cancel: UIAlertAction = UIAlertAction(title: "Cancel",
-                                                              style: .cancel,
-                                                              handler: { (action) in
-                        self.toggleUIState(isEnabled: true, showCallControl: false)
-                        self.stopSpin()
-                    })
-                    alertController.addAction(cancel)
-                    
-                    self.present(alertController, animated: true, completion: nil)
-                } else {
-                    self.performStartCallAction(uuid: uuid, handle: handle)
-                }
+            return
+        }
+        
+        let uuid = UUID()
+        let handle = "Voice Bot"
```

Comment:
Looks like these could be moved inside the block below because they are not used outside of it.